src/veryfi/Client.php

```diff
@@ -550,4 +551,297 @@ public function replace_tags(int $document_id,
+     */
+    public function process_w2_document(string $file_path,
+                                        bool $delete_after_processing = false,
+                                        int $max_pages_to_process = 1,
+                                        array $additional_request_parameters = array()): string
+    {
+        $endpoint_name = '/w2s/';
+        $file_name = pathinfo($file_path, PATHINFO_BASENAME);
+        $file = fopen($file_path, 'rb');
+        $file_data = base64_encode(fread($file, filesize($file_path)));
+        $request_arguments = array(
+            'file_name' => $file_name,
+            'file_data' => $file_data,
+            'auto_delete' => $delete_after_processing,
+            'max_pages_to_process' => $max_pages_to_process
+        );
+        $request_arguments = array_replace($request_arguments, $additional_request_parameters);
+        return $this->request('POST', $endpoint_name, $request_arguments);
+    }
+
+    /**
+     * Process a W2 document from a URL and extract all fields from it.
+     *
+     * @param string $file_name The file name including the extension
+     * @param string $file_url Publicly accessible URL to a file
+     * @param array|null $file_urls List of publicly accessible URLs to multiple files
+     * @param boolean $delete_after_processing Delete this document from Veryfi after data has been extracted
+     * @param int $max_pages_to_process The number of pages to process for the document
+     * @param array $additional_request_parameters Additional request parameters
+     * @return string Data extracted from the document
+     */
+    public function process_w2_document_from_url(string $file_name, string $file_url, array $file_urls = null, bool $delete_after_processing = false, int $max_pages_to_process = 1, array $additional_request_parameters = []): string
+    {
+        $endpoint_name = "/w2s/";
+        $request_arguments = array_merge([
+            'file_name' => $file_name,
+            'auto_delete' => $delete_after_processing,
+            'file_url' => $file_url,
+            'file_urls' => $file_urls,
+            'max_pages_to_process' => $max_pages_to_process
+        ], $additional_request_parameters);
+
+        return $this->request("POST", $endpoint_name, $request_arguments);
+    }
+
+    /**
+     * Process any document and extract all the fields from it.
+     *
+     * @param string $file_path Path on disk to a file to submit for data extraction
+     * @param string $template_name The name of the extraction template
+     * @param int $max_pages_to_process The number of pages to process for the document
+     * @param array $additional_request_parameters Additional request parameters
+     * @return string Data extracted from the document
+     */
+    public function process_any_document_from_file(string $file_path, string $template_name, int $max_pages_to_process = 20, array $additional_request_parameters = []): string
```

Comment:
how it's different from process_any_document?

Reply:
1.	File Name Extraction:
	•	First Snippet: Uses basename($file_path) to extract the file name.
	•	Second Snippet: Uses pathinfo($file_path, PATHINFO_BASENAME) to extract the file name.
Effect: Both methods return the base name of the file, so functionally they are identical. However, pathinfo() is more versatile as it can extract other file path components (like file extension) if needed.
	2.	File Handling:
	•	First Snippet: Uses file_get_contents($file_path) to read the file in one go and encode it in base64.
	•	Second Snippet: Uses fopen() to open the file, then reads it using fread(), and encodes the content in base64.
Effect:
	•	file_get_contents() is simpler and reads the entire file at once. It’s easier to use, but can consume more memory for very large files since it loads everything into memory at once.
	•	fopen() with fread() is more granular and might give you more control over memory usage. However, for very large files, this approach can be more efficient as it allows for buffered reading (although here the entire file is still read into memory).

Reply:
Done I removed the function who uses  file_get_contents() because it can consume more memory for very large files since it loads everything into memory at once